docs/asciidoc/http-client.adoc

```diff
@@ -512,3 +518,188 @@ include::{examplesdir}/resolver/custom/Application.java[lines=18..38]
+==== Connection Timeout
+The following listing shows all available connection timeout configuration options, but some of them may apply only to a specific transport.
+
+* `CONNECT_TIMEOUT_MILLIS` - if the connection establishment attempt to the remote peer does not finish within the configured connect timeout (resolution: ms),
+the connection establishment attempt will be failed. Default: 30s.
+* `SO_KEEPALIVE` - when the connection stays idle for some time (the time is implementation dependent, but typically 2 hours),
+TCP automatically sends a `keepalive` probe to the remote peer.
+By default, https://docs.oracle.com/javase/8/docs/api/java/net/SocketOptions.html#SO_KEEPALIVE[SO_KEEPALIVE] is not enabled.
+When you run with `Epoll` transport you may also configure:
+** `TCP_KEEPIDLE` - the maximum time (resolution: seconds) that this connection stays idle before TCP starts sending `keepalive` probes,
+if `SO_KEEPALIVE` has been set. The maximum time is implementation dependent, but typically the default is 2 hours.
+** `TCP_KEEPINTVL` - the time (resolution: seconds) between individual `keepalive` probes.
+** `TCP_KEEPCNT` - the maximum number of `keepalive` probes TCP should send before dropping the connection.
+
+NOTE: Sometimes between the client and the server you may have a network component that silently drops the idle connections without sending a response.
+From Reactor Netty point of view in this use case the remote peer just does not respond.
+In order to be able to handle such use case you may consider configuring
+https://docs.oracle.com/javase/8/docs/technotes/guides/net/socketOpt.html[SO_KEEPALIVE].
+
+To customize the default settings, you can configure `HttpClient` as follows:
+
+====
+[source,java,indent=0]
+.{examplesdir}/channeloptions/Application.java
+----
+include::{examplesdir}/channeloptions/Application.java[lines=18..46]
+----
+<1> Configures the connection establishment timeout to 10 seconds.
+<2> Enables TCP `keepalive`. This means that TCP starts sending `keepalive` probes when a connection is idle for some time.
+<3> The connection needs to remain idle 5 minutes before TCP starts sending `keepalive` probes.
+<4> Configures the time between individual `keepalive` probes to 1 minute.
+<5> Configures the maximum number of TCP `keepalive` probes to 8.
+====
+
+[[ssl-tls-timeout]]
+==== SSL/TLS Timeout
+`HttpClient` supports the SSL/TLS functionality provided by Netty.
+
+The below shows the available timeout configuration options:
+
+* `handshakeTimeout` - this option can be used for configuring SSL handshake timeout (resolution: ms). Default: 10s.
+
+NOTE: You should consider increasing the SSL handshake timeout when expecting slow network connections.
+
+* `closeNotifyFlushTimeout` - this option can be used for configuring SSL `close_notify` flush timeout (resolution: ms). Default: 3s.
+* `closeNotifyReadTimeout` - this option can be used for configuring SSL `close_notify` read timeout (resolution: ms). Default: 0s.
```

Comment:
Change "this option can be used for configuring SSL" to "Use this option to configure the SSL".